pkg/clientset/clientset.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientset
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/andygrunwald/go-jira"
+	"github.com/prometheus-community/jiralert/pkg/config"
+)
+
+type ClientSet struct {
+	jira map[string]*jira.Client
+	sync.RWMutex
+}
+
+var ErrorClientExists = errors.New("client already exists")
+
+func (c *ClientSet) GetJira(userName string) (*jira.Client, bool) {
+	c.RLock()
+	jc, ok := c.jira[userName]
+	c.RUnlock()
+
+	return jc, ok
+}
+
+func (c *ClientSet) NewJira(conf *config.ReceiverConfig) (*jira.Client, error) {
```

Comment:
Can we reduce the scope of public APIs? Sounds like we have 3 methods and 1 `ErrorClientExists` but we only need `GetOrCreateJira`, no? We can always make something public later (:

Reply:
sure, I've updated the PR.